pkg/controller/workload/job/job_webhook.go

```diff
@@ -0,0 +1,112 @@
+limitations under the License.
+*/
+
+package job
+
+import (
+	"context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+)
+
+// log is for logging in this package.
+var webhookLog = ctrl.Log.WithName("job-webhook")
+
+type JobWebhook struct {
+	manageJobsWithoutQueueName bool
+}
+
+// SetupWebhook configures the webhook for batchJob.
+func SetupWebhook(mgr ctrl.Manager, opts ...Option) error {
+	options := defaultOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	wh := &JobWebhook{
+		manageJobsWithoutQueueName: options.manageJobsWithoutQueueName,
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&batchv1.Job{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-batch-v1-job,mutating=true,failurePolicy=fail,sideEffects=None,groups=batch,resources=jobs,verbs=create,versions=v1,name=mjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &JobWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *JobWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*batchv1.Job)
+	webhookLog.V(5).Info("Applying defaults", "job", klog.KObj(job))
+
+	if queueName(job) == "" && w.manageJobsWithoutQueueName && !(*job.Spec.Suspend) {
+		job.Spec.Suspend = pointer.Bool(true)
+		return nil
+	}
+
+	if queueName(job) != "" && !(*job.Spec.Suspend) {
+		job.Spec.Suspend = pointer.Bool(true)
+		return nil
+	}
+
+	return nil
+}
+
+// +kubebuilder:webhook:path=/validate-batch-v1-job,mutating=false,failurePolicy=fail,sideEffects=None,groups=batch,resources=jobs,verbs=update,versions=v1,name=vjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &JobWebhook{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (w *JobWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	return nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (w *JobWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
+	oldJob := oldObj.(*batchv1.Job)
+	newJob := newObj.(*batchv1.Job)
+	webhookLog.V(5).Info("Validating update", "job", klog.KObj(newJob))
+
+	return validateUpdate(oldJob, newJob, w.manageJobsWithoutQueueName)
+}
+
+func validateUpdate(oldJob, newJob *batchv1.Job, manageJobsWithoutQueueName bool) error {
+	suspendPath := field.NewPath("job", "spec", "suspend")
+
+	if !manageJobsWithoutQueueName && queueName(oldJob) == "" && queueName(newJob) != "" && !*newJob.Spec.Suspend {
+		return field.Required(suspendPath, "suspend should be true when adding the queue name with manageJobsWithoutQueueName set to false")
```

Comment:
I guess it shouldn't matter here what's the value of `manageJobsWithoutQueueName`?

Because you are adding a queue, so it will be managed now.

Reply:
This refers to the specified scenario that, manageJobsWithoutQueueName is not enabled, user create a job without a queueName, then he wants to manage this by kueue, so he adds a queueName. At that moment, suspend is not allowed. When `manageJobsWithoutQueueName` enabled, it will be captured by below `queueName(oldJob) != queueName(newJob)`.

But what you said also makes sense, updated as advised.